### PR TITLE
Improve OptimizeCrystalplacementTest

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

The [incremental master](http://builds.mantidproject.org/job/master_incremental/label=ubuntu-16.04-build/lastCompletedBuild/testReport/) build on Ubuntu 16.04 is unstable w.r.t  `OptimizeCrystalplacementTest`, while the clean build seems unaffected. This is an attempt to reduce the coupling between the test methods and reduce the usage of the ADS in a hope that the issues is the global state being corrupted when the tests are run in parallel.

This is a shot in the dark but the code has been improved nonetheless.

**To test:**

Review the code carefully and ensure all tests still pass. 

No issue

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h [hunk 1]
```diff
@@ -47,223 +73,174 @@ class OptimizeCrystalPlacementTest : public CxxTest::TestSuite {
...
-    table = alg1.getProperty("FitInfoTable");
-
-    Geometry::Goniometer IGon(origGon5638);
-    std::vector<double> GonAngles5638 = IGon.getEulerAngles("YZY");
-
-    for (size_t i = 0; i < table->rowCount(); ++i) {
-      std::string nm = table->String(i, 0);
+    }
+    Goniometer instGoniom(goniomMat);
+    std::vector<double> chiphiOmega = instGoniom.getEulerAngles("YZY");
```
Comment (soininen) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: My IDE is of the opinion that `chiphiOmega` is unused.

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h [hunk 2]
```diff
@@ -47,223 +73,174 @@ class OptimizeCrystalPlacementTest : public CxxTest::TestSuite {
...
+    auto resultsFix5637 = runOptimizePlacement(
+        modPeaksNoFix, {{"KeepGoniometerFixedfor", "5637"}});
+    auto fitInfoFix5637 = resultsFix5637.second;
+    const auto angles5638 = Goniometer(origGon5638Mat).getEulerAngles("YZY");
+
+    for (size_t i = 0; i < fitInfoFix5637->rowCount(); ++i) {
+      std::string nm = fitInfoFix5637->String(i, 0);
```
Comment (soininen) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: I think `nm` could be declared `const`.

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h [hunk 3]
```diff
@@ -47,223 +73,174 @@ class OptimizeCrystalPlacementTest : public CxxTest::TestSuite {
...
-      if (RunNum == 5637) {
-        if (origGon5637 == ZMat)
-          origGon5637 = peak.getGoniometerMatrix();
+    Matrix<double> origGon5637Mat;
+    for (int i = 0; i < npeaks; ++i) {
+      auto &peak = modPeaksNoFix->getPeak(i);
+      const int runNum = peak.getRunNumber();
 
-        peak.setGoniometerMatrix(tilt * origGon5637);
+      Matrix<double> GG;
```
Comment (soininen) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: `GG` doesn't seem to be used anywhere. Or at least that's what my IDE says.

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h [hunk 4]
```diff
@@ -47,223 +73,174 @@ class OptimizeCrystalPlacementTest : public CxxTest::TestSuite {
...
     }
 
-    Geometry::Goniometer IGon(origGon5638);
-    std::vector<double> GonAngles5638 = IGon.getEulerAngles("YZY");
-
-    Geometry::Goniometer IGon2(origGon5637);
-    std::vector<double> GonAngles5637 = IGon2.getEulerAngles("YZY");
+    const auto angles5638 = Goniometer(origGon5638Mat).getEulerAngles("YZY");
+    const auto angles5637 = Goniometer(origGon5637Mat).getEulerAngles("YZY");
+    auto resultsTiltFixBoth = runOptimizePlacement(
```
Comment (soininen) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: Could `resultsTiltFixBoth` be declared `const`?

**Review by soininen — CHANGES_REQUESTED** [review]

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h (hunk 1 above)
Comment (martyngigg) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: Mine usually tells me that too but you're right it's not used anywhere.

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h (hunk 2 above)
Comment (martyngigg) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: :+1:

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h (hunk 3 above)
Comment (martyngigg) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: :+1:

File: Framework/Crystal/test/OptimizeCrystalPlacementTest.h (hunk 4 above)
Comment (martyngigg) [inline Framework/Crystal/test/OptimizeCrystalPlacementTest.h]: Indeed it can :+1:

**Review by soininen — APPROVED** [review]

**Comment by soininen** [comment]
The changes look sensible and all tests passed. Hopefully this will help with the incremental master build :shipit:

**Pull request merged**